src/Launcher.cpp

```diff
@@ -155,159 +176,251 @@ int Launcher::automaticMerge() {
 
 void Launcher::parseCommandLine() {
     auto tr = [&] (const char * text) { return QCoreApplication::translate("Help", text); };
-    for (int i = 1; i < argc; ++i) {
-        if (string("-o") == argv[i]) {
-            if (++i < argc) {
-                saveOptions.fileName = argv[i];
-            }
-        } else if (string("-m") == argv[i]) {
-            if (++i < argc) {
-                saveOptions.maskFileName = argv[i];
-                saveOptions.saveMask = true;
-            }
-        } else if (string("-v") == argv[i]) {
+
+    QCommandLineParser parser;
+    parser.setApplicationDescription(
+            tr("Merges raw_files into an HDR DNG raw image.\n") +
+#ifndef NO_GUI
+            tr("If neither -o nor --group options are given, the GUI will be presented.\n") +
+#endif
+            tr("If similar options are specified, only the last one prevails.\n")
+    );
+
+    parser.addPositionalArgument("files", "The input raw files", "[raw_files...]");
+
+    QCommandLineOption outputOpt("o",
+            tr("Sets <file> as the output file name. "
+            "If not set it falls back to '%1'.\n"
+            "The following parameters are accepted, most useful in batch mode:").arg("%id[-1]/%iF[0]-%in[-1].dng") +
+            "\n- %if[n]: " + tr("Replaced by the base file name of image n. Image file names "
+            "are first sorted in lexicographical order. Besides, n = -1 is the "
+            "last image, n = -2 is the previous to the last image, and so on.") +
+            "\n- %iF[n]: " + tr("Replaced by the base file name of image n without the extension.") +
+            "\n- %id[n]: " + tr("Replaced by the directory name of image n.") +
+            "\n- %in[n]: " + tr("Replaced by the numerical suffix of image n, if it exists."
+            "For instance, in IMG_1234.CR2, the numerical suffix would be 1234.") +
+            "\n- %%: " + tr("Replaced by a single %."),
+            "file");
+    parser.addOption(outputOpt);
+
+    QCommandLineOption maskFileOpt("m",
+            tr("Saves the mask to <file> as a PNG image.\n"
+            "Besides the parameters accepted by -o, it also accepts:") +
+            "\n- %of: " + tr("Replaced by the base file name of the output file.") +
+            "\n- %od: " + tr("Replaced by the directory name of the output file."),
+            "file");
+    parser.addOption(maskFileOpt);
```

Comment:
It's not necessary, the handler chould be changed to recieve strings. But of course it's easier to use and less prone to copy/paste errors. (passing "-s" to the handler when it should be "-g"...)
I used addOptions() with the C++11 curly bracket initializers as shown in the docs:
```
    parser.addOptions({
        // A boolean option with a single name (-p)
        {"p",
            QCoreApplication::translate("main", "Show progress during copy")},
        // A boolean option with multiple names (-f, --force)
        {{"f", "force"},
            QCoreApplication::translate("main", "Overwrite existing files.")},
        // An option with a value
        {{"t", "target-directory"},
            QCoreApplication::translate("main", "Copy all source files into <directory>."),
            QCoreApplication::translate("main", "directory")},
    });
```
But I switched to explicit objetcs as 1) parser.isSet(optionInstance) is the recommended way to check for options without value and 2) it is safer to use objects rather than strings in parser.isSet() and parser.value(). Being able to use it in the handler was a bonus :)

I will make it const, and all the other QStrings, autos, ... too.